CMakeLists.txt

```diff
@@ -562,6 +562,32 @@ if ( INSTALL_ZKG )
         DESTINATION ${ZEEK_ZKG_CONFIG_DIR} RENAME config)
 endif ()
 
+########################################################################
+## Look for external plugins to build in
+
+if( NOT ("${ZEEK_BUILTIN_PLUGINS_DIR}" STREQUAL "") AND IS_DIRECTORY ${ZEEK_BUILTIN_PLUGINS_DIR})
+    message (STATUS "Searching for external plugins in ${ZEEK_BUILTIN_PLUGINS_DIR}")
+    file(GLOB PLUGIN_DIRS CONFIGURE_DEPENDS "${ZEEK_BUILTIN_PLUGINS_DIR}/*/CMakeLists.txt")
```

Comment:
Yeah, you're right about the behavior with old versions of cmake.  I didn't bother documenting it because I felt like it was fairly low risk of issues since someone could just rerun configure explicitly to make that update (I think?).  I would like to avoid putting more explanation in the configure help output.  Would it make more sense to put more explanation in documentation somewhere?  We could mention that there are some caveats to the feature and point to the docs

I've been using ninja and it seems to be working fine there.  I can check make as well.  I found some references to problems with other generators, but I didn't see any reference to that in the CMake docs.

One option we have is that we could stop using glob by forcing users to specify a list of directories where plugins can be found for the configure step.  We would then just add each of those subdirectories to the build instead of doing the glob.  What do you think about that?

Reply:
Ok, I might implement a change to the configure/cmake option.  Maybe something like this...

```
--build-in-plugin=~/my-plugin --build-in-plugin=/whatever/zeek-af-packet
```

I like this model because it forces users to be much more explicit about what they want to build and it completely avoids the potential issues that could crop up from the glob.

Reply:
Generally think it's worth avoiding the vagueness around whether `CONFIGURE_DEPENDS` will work in all cases.

Explicit list in the configure flags seems fine and probably simplest way.  Only other idea is to instead point to a manifest file and include it so that CMake will re-run and regenerate build system whenever it's updated.  Example patch at https://github.com/zeek/zeek/commit/8fd81ea719bcb0293a922a824d80a0c67d15cd2c

Manifest file looks like this:

```
$ cat manifest.cmake
zeek_add_builtin_plugin("/home/jon/tmp/zeek-plugins/zeek-af_packet-plugin")
zeek_add_builtin_plugin("/home/jon/tmp/zeek-plugins/zeek-quic")
```

But the explicit list of `--build-in-plugin=` like you said may be simplest if you want to go with that.

Reply:
Sorry I haven't taken care of this yet, I'm still planning on getting this all finished soon.